src/Topologies/grids.jl

```diff
@@ -1,14 +1,15 @@
+
 """
-    GridTopology(n1,n2)
+TensorProductTopology(mesh)
```

Comment:
Should be indented 4 spaces so it renders as a code block in markdown
```suggestion
    TensorProductTopology(mesh)
```

Reply:
My main concern with the name `TensorProductTopology` is that we are going to use the term "tensor" elsewhere for a very different object (`AxisTensor`). I don't want to get too bogged down in naming stuff (we can always rename things later), so it don't let it block this, but do you have any other ideas (would `ProductTopology` be sufficient?)

Reply:
And perhaps rename this file accordingly (e.g `tensorproduct.jl`/`product.jl`)?

Reply:
Thanks for the feedback. No, unfortunately the term `ProductTopology` would not mean anything. How about `RegularGrid` and `IrregularGrid`? Or `UniformGrid` and `NonUniformGrid`?

Reply:
In [`82f89ed`](https://github.com/CliMA/ClimaCore.jl/pull/82/commits/82f89ed04ce33546240e0489473b174aad391484), I renamed back the original `GridTopology`, as it was, and put all implementations (dispatching on either type of concrete Mesh) in the same original file `grids.jl`. I hope it is clearer now. Thanks